rqt_py_common/src/rqt_py_common/topic_completer.py

```diff
@@ -95,52 +95,54 @@ def update_topics(self, node):
         QTreeView, QVBoxLayout, QWidget
 
     import rclpy
-    rclpy.init()
-    topic_completer_node = rclpy.create_node()
-
-    app = QApplication(sys.argv)
-    mw = QMainWindow()
-    widget = QWidget(mw)
-    layout = QVBoxLayout(widget)
-
-    edit = QLineEdit()
-    edit_completer = TopicCompleter(edit)
-    edit_completer.update_topics(topic_completer_node)
-    # edit_completer.setCompletionMode(QCompleter.InlineCompletion)
-    edit.setCompleter(edit_completer)
-
-    combo = QComboBox()
-    combo.setEditable(True)
-    combo_completer = TopicCompleter(combo)
-    combo_completer.update_topics(topic_completer_node)
-
-    # combo_completer.setCompletionMode(QCompleter.InlineCompletion)
-    combo.lineEdit().setCompleter(combo_completer)
-
-    model_tree = QTreeView()
-    model_tree.setModel(combo_completer.model())
-    model_tree.expandAll()
-    for column in range(combo_completer.model().columnCount()):
-        model_tree.resizeColumnToContents(column)
-
-    completion_tree = QTreeView()
-    completion_tree.setModel(combo_completer.completionModel())
-    completion_tree.expandAll()
-    for column in range(combo_completer.completionModel().columnCount()):
-        completion_tree.resizeColumnToContents(column)
-
-    layout.addWidget(model_tree)
-    layout.addWidget(completion_tree)
-    layout.addWidget(edit)
-    layout.addWidget(combo)
-    layout.setStretchFactor(model_tree, 1)
-    widget.setLayout(layout)
-    mw.setCentralWidget(widget)
-
-    mw.move(300, 0)
-    mw.resize(800, 900)
-    mw.show()
-    app.exec_()
-
-    topic_completer_node.destroy_node()
-    rclpy.shutdown()
+    from rclpy.executors import ExternalShutdownException
+
+    try:
```

Comment:
> Nit on the scope of this try/except since it's a bigger chunk of code. I think the `try/except` only needs to surround the part that blocks, which IIUC is the `app.exec_()`. I'd recommend reducing it to just that in this case.

I've been opting for surrounding most/all of the code in the try/except block.  The basic reason for that is to capture the `KeyboardInterrupt` everywhere.  That is, if you happen to be in the "setup" phase (before the `app.exec_()`, and you hit Ctrl-C, then you would still get an ugly exit with the raise of the `KeyboardInterrupt` exception.  While this is fairly unlikely, by surrounding everything we make sure that we always have a "clean" exit.

That said, I don't feel that strongly about it, so if you think it is better to reduce the scope I can go ahead and do that.

Reply:
I don't feel strongly about it either. Nothing really bad happens from putting too much code into a try/except block that catches specific exceptions. It just hides what code is actually expected to raise